src/librustdoc/clean/types.rs

```diff
@@ -648,13 +649,16 @@ impl Attributes {
                         if let Some(ref fragment) = *fragment {
                             let cache = cache();
                             let url = match cache.extern_locations.get(krate) {
-                                Some(&(_, ref src, ExternalLocation::Local)) => {
-                                    src.to_str().expect("invalid file path")
-                                }
-                                Some(&(_, _, ExternalLocation::Remote(ref s))) => s,
-                                Some(&(_, _, ExternalLocation::Unknown)) | None => {
-                                    "https://doc.rust-lang.org/nightly"
+                                Some(&(ref krate_name, _, ExternalLocation::Local))
+                                    if krate_name == "core" =>
```

Comment:
Since we see the same behavior both ways, let's remove this special-casing to make things easier when we end up fixing the `#[lang]` issue.